torch/distributed/launch.py

```diff
@@ -232,6 +236,10 @@ def main():
                    args.training_script,
                    "--local_rank={}".format(local_rank)] + args.training_script_args
 
+        # insert the -m flag for module launching
+        if args.module:
+            cmd.insert(2, "-m")
+
```

Comment:
Since the `cmd` is initialized in the same way in both code paths above, could you move the `-m` above as well? Having an `insert` here seems brittle and can easily break if more args are added above later.

I'm thinking (untested):

```python
cmd = [sys.executable, "-u"]

if args.module:
  cmd.append("-m")

cmd.append(args.training_script)

if not args.use_env:
  cmd.append("--local_rank={}".format(local_rank))

cmd.extend(args.training_script_args)
```